consensus/polybft/fsm.go

```diff
@@ -535,15 +541,23 @@ func (f *fsm) Insert(p *pbft.SealedProposal) error {
 	// Write extar data to header
 	f.block.Block.Header.ExtraData = append(make([]byte, 32), extra.MarshalRLPTo(nil)...)
 
-	if err := f.backend.CommitBlock(f.block); err != nil {
+	// commit exit events only when we finalize a block
+	events, err := getExitEventsFromReceipts(f.epochNumber, f.block.Block.Number(), f.blockBuilder.Receipts())
```

Comment:
This is not correct: only proposer would have up to date `blockBuilder` right? Therefore you won't have receipts for non-proposer nodes...?

If I'm correct, you would need to execute transactions here in order to have receipts.

Reply:
Yes, it's a copy/paste error. Good catch :100: . It should be f.block.Receipts there.

Fix: https://github.com/0xPolygon/polygon-edge/pull/852/commits/2905c39f46bf6a7af6d323b1e016924bb55a3573